CRM/Volunteer/APIWrapper/CustomField.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+class CRM_Volunteer_APIWrapper_CustomField implements API_Wrapper {
+
+  /**
+   * Interpret and alter the API request as needed.
+   *
+   * @param array $apiRequest
+   */
+  public function fromApiInput($apiRequest) {
+    // nothing to do here
+    return $apiRequest;
+  }
+
+  /**
+   * Modify the API result before returning it.
+   *
+   * If the request's action is whitelisted and 'custom' is specified as a
+   * return, add all custom field values to the result.
+   *
+   * @param array $apiRequest
+   * @param array $result
+   */
+  public function toApiOutput($apiRequest, $result) {
```

Comment:
comments considered harmful: tip for being an eXtreme programmer: write a function, "returnAllCustomFields()" and call it after you've validated the prerquisites in this "hook" (I know it's not a hook).